yajsapi/executor/executor.ts

```diff
@@ -376,35 +378,24 @@ export class TaskExecutor {
     this.options.eventTarget?.dispatchEvent(new Events.ComputationFailed({ reason: e.toString() }));
     this.logger?.error(e.toString());
     if (this.isRunning) this.logger?.warn("Trying to stop executor...");
-    this.end().catch((e) => {
-      this.logger?.error(e);
-      if (runtimeContextChecker.isNode && !this.options.isSubprocess) process?.exit(1);
-    });
+    this.end().catch((e) => this.logger?.error(e));
   }
 
   private handleCancelEvent() {
-    terminatingSignals.forEach((event) => process.on(event, this.cancel.bind(this)));
+    terminatingSignals.forEach((event) => process.on(event, () => this.cancel(event)));
   }
 
   private removeCancelEvent() {
-    terminatingSignals.forEach((event) => process.off(event, this.cancel.bind(this)));
+    terminatingSignals.forEach((event) => process.removeAllListeners(event));
   }
 
   public async cancel(reason?: string) {
     if (this.isCanceled) return;
-    if (runtimeContextChecker.isNode && !this.options.isSubprocess) this.removeCancelEvent();
+    if (runtimeContextChecker.isNode) this.removeCancelEvent();
     const message = `Executor has interrupted by the user. Reason: ${reason}.`;
     this.logger?.warn(`${message}. Stopping all tasks...`);
     this.isCanceled = true;
-    await this.end()
-      .then(() => {
-        if (this.options.isSubprocess) throw new Error(message);
-        else if (runtimeContextChecker.isNode) process.exit(0);
-      })
-      .catch((error) => {
-        this.logger?.error(error);
-        if (runtimeContextChecker.isNode && !this.options.isSubprocess) process.exit(1);
-      });
+    await this.end().catch((error) => this.logger?.error(error));
```

Comment:
Please don't mix `await` and `catch` like that - it's mixing coding styles and making things harder to read, understand and maintain. Instead consider:

```ts
  public async cancel(reason?: string) {
    try {
      if (this.isCanceled) return;
      if (runtimeContextChecker.isNode) this.removeCancelEvent();
      const message = `Executor has interrupted by the user. Reason: ${reason}.`;
      this.logger?.warn(`${message}. Stopping all tasks...`);
      this.isCanceled = true;
      await this.end();
    } catch (err) {
      this.logger?.error(err, "Error while cancelling the executor");
    }
  }
```

Reply:
yes you may be right i will change it here but unfortunately we have it in many places...

Reply:
One step at a time.